code/game/turfs/simulated/floor_acts.dm

```diff
@@ -1,27 +1,15 @@
+//EX act uses new damage type blast, which allows it to get burn AND damage overlays simultaneously
 /turf/simulated/floor/ex_act(severity)
 	//set src in oview(1)
 	switch(severity)
 		if(1.0)
-			src.ChangeTurf(get_base_turf_by_area(src))
+			take_damage(rand(350, 600), BLAST) //Breaks through 3 - 4 layers
 		if(2.0)
-			switch(pick(40;1,40;2,3))
```

Comment:
byebye to all this crap logic. Ex act just does damage now, and the take_damage function handles breaking down